### PR TITLE
haproxy: add plist file and caveats

### PR DESCRIPTION
**Pull request by aviddiviner**
- [✓] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [✓] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [✓] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [✓] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Yep, I think I've followed all the steps. Also tested that the plist file works nicely, and if things don't work then haproxy does log its errors and whatnot.

Let me know what you folks think. Thanks for this awesome piece of software!

### CODE REVIEW

File: Formula/haproxy.rb [hunk 1]
```diff
@@ -32,6 +32,40 @@ def install
...
 
+  def caveats; <<~EOS
+    To start haproxy with brew services, first create and edit your config at:
+      #{etc}/haproxy.cfg
+    Logs can be found at:
+      #{var}/log/haproxy.log
+    EOS
+  end
+
+  plist_options :manual => "haproxy"
```
Comment (ilovezfs) [inline Formula/haproxy.rb]: should this also have the -f?

File: Formula/haproxy.rb [hunk 2]
```diff
@@ -32,6 +32,40 @@ def install
     bin.install "haproxy"
   end
 
+  def caveats; <<~EOS
```
Comment (ilovezfs) [inline Formula/haproxy.rb]: these caveats aren't needed

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/haproxy.rb (hunk 1 above)
Comment (aviddiviner) [reply Formula/haproxy.rb]: Yea, let me rather just remove this.

File: Formula/haproxy.rb (hunk 2 above)
Comment (aviddiviner) [reply Formula/haproxy.rb]: I'd like to at least let the user know that it won't work without a config file at `/usr/local/etc/haproxy.cfg`.

Is there some easy way I can include my own sample config in the package?

File: Formula/haproxy.rb (hunk 1 above)
Comment (aviddiviner) [reply Formula/haproxy.rb]: Ah, I see, that's a good idea, let me add this.

File: Formula/haproxy.rb (hunk 2 above)
Comment (aviddiviner) [reply Formula/haproxy.rb]: Actually, with the `-f path/to/config` added to `plist_options` it's quite clear to the user which config file it's using, so I agree, I can remove these caveats.

**Comment by aviddiviner** [comment]
I'm going to close this PR and open a new one, sorry. I didn't branch, and I messed it up now :P

**Pull request closed without merging**